owners/test/reviewer_selection.test.js

```diff
@@ -0,0 +1,253 @@
+        kid: ['biz/style.css'],
+        thirdChild: ['buzz/info.txt', 'buzz/code.js'],
+      });
+
+      expect(reviewerFiles).toEqual(
+        expect.arrayContaining([
+          ['child', ['foo/file.js', 'foo/bar/other_file.js']],
+          ['thirdChild', ['buzz/info.txt', 'buzz/code.js']],
+        ])
+      );
+    });
+  });
+
+  describe('pickBestReview', () => {
+    const fileTreeMap = ReviewerSelection.buildFileTreeMap(
+      [
+        './main.js', // root
+        'biz/style.css', // child, kid, root
+        'foo/bar/other_file.js', // child, root
+        'buzz/info.txt', // thirdChild, root
+        'buzz/code.js', // thirdChild, root
+      ],
+      ownersTree
+    );
+
+    it('builds a map from deepest reviewers to files they own', () => {
+      sandbox.stub(ReviewerSelection, '_reviewersWithMostFiles').callThrough();
+      ReviewerSelection._pickBestReview(fileTreeMap);
+
+      sandbox.assert.calledWith(ReviewerSelection._reviewersWithMostFiles, {
+        child: ['biz/style.css', 'foo/bar/other_file.js'],
+        kid: ['biz/style.css'],
+        thirdChild: ['buzz/info.txt', 'buzz/code.js'],
+      });
+    });
+
+    it('picks one of the reviewers with the most files', () => {
+      const [bestReviewer, filesCovered] = ReviewerSelection._pickBestReview(
+        fileTreeMap
+      );
+
+      expect(['child', 'thirdChild']).toContain(bestReviewer);
+      expect(filesCovered.length).toEqual(2);
+    });
+  });
+
+  describe('buildFileTreeMap', () => {
+    it('builds a map from filenames to subtrees', () => {
+      const fileTreeMap = ReviewerSelection.buildFileTreeMap(
+        [
+          './main.js',
+          './package.json',
+          'biz/style.css',
+          'foo/file.js',
+          'foo/bar/other_file.js',
+          'buzz/info.txt',
+          'buzz/code.js',
+        ],
+        ownersTree
+      );
+
+      expect(fileTreeMap).toEqual({
+        './main.js': rootDirTree,
+        './package.json': rootDirTree,
+        'biz/style.css': otherChildDirTree,
+        'foo/file.js': childDirTree,
+        'foo/bar/other_file.js': childDirTree,
+        'buzz/info.txt': thirdChildDirTree,
+        'buzz/code.js': thirdChildDirTree,
+      });
+    });
+  });
+
+  describe('pickReviewers', () => {
+    let fileTreeMap;
+    let bestReviewerStub;
+
+    beforeEach(() => {
+      bestReviewerStub = sandbox.stub(ReviewerSelection, '_pickBestReview');
+      fileTreeMap = ReviewerSelection.buildFileTreeMap(
+        ['./main.js', 'foo/file.js', 'biz/style.css', 'buzz/info.txt'],
+        ownersTree
+      );
+    });
+
+    it('picks reviewers until all files are covered', () => {
+      bestReviewerStub.callThrough();
+      const reviews = ReviewerSelection.pickReviewers(fileTreeMap);
+      const reviewers = reviews.map(([reviewer, files]) => reviewer);
+
+      sandbox.assert.calledThrice(ReviewerSelection._pickBestReview);
+      expect(reviewers).toEqual(['child', 'thirdChild', 'root']);
+    });
+
```

Comment:
Add a test that shows this doesn't necessarily add the root user even if there are multiple potential owners

Reply:
Done